storage/oasis/consensus.go

```diff
@@ -85,6 +85,12 @@ func (cc *ConsensusClient) AllData(ctx context.Context, height int64) (*storage.
 	return &data, nil
 }
 
+// Implements ConsensusSourceStorage.
+func (cc *ConsensusClient) HasBlock(ctx context.Context, height int64) error {
+	_, err := cc.nodeApi.GetBlock(ctx, height)
```

Comment:
I think having the option to fetch the latest available height will already be helpful now with the "mega-query".

So that the analyzer doesn't pick up and lock up to 100 blocks that are after the latest height that it has available.

Reply:
:+1: if we have N analyzers running, they would actually lock as many as 100*N blocks above the latest height, collectively. OTOH we only expect to be running parallel analyzers on old blocks, not around latest height.

Reply:
Replaced `HasBlock` with `LatestBlockHeight` method.